clients/roscpp/include/ros/service_server_link.h

```diff
@@ -37,6 +37,21 @@
 
 #include "ros/common.h"
 
+// check if we might need to include our own backported version boost::condition_variable
```

Comment:
This should be added at the very beginning, before the include of "ros/common.h" to make sure that boost thread/condition_variable is not accidentally included before...

Reply:
Done.

The test is passing locally to me all time. Hopefully it was just this in the buildfarm.
Or maybe it's something related with the `wait_for` + `nanoseconds` changed ?!

There are also the md5sum mismatch errors before (http://build.ros.org/job/Lpr__ros_comm__ubuntu_xenial_amd64/679/consoleFull), but I also have them locally, so it's probably part of some test ?!

``` bash
~/dev/ws/cpr_ws/build/test_roscpp$ make run_tests_test_roscpp_rostest_test_launch_service_call.xml
-- run_tests.py: execute commands
  /home/eperdomo/dev/ws/cpr_ws/src/ros_comm/tools/rostest/scripts/rostest --pkgdir=/home/eperdomo/dev/ws/cpr_ws/src/ros_comm/test/test_roscpp --package=test_roscpp --results-filename test_launch_service_call.xml --results-base-dir /home/eperdomo/dev/ws/cpr_ws/build/test_roscpp/test_results /home/eperdomo/dev/ws/cpr_ws/src/ros_comm/test/test_roscpp/test/launch/service_call.xml 
... logging to /home/eperdomo/.ros/log/rostest-nb-eperdomo-19190.log
[ROSUNIT] Outputting test results to /home/eperdomo/dev/ws/cpr_ws/build/test_roscpp/test_results/test_roscpp/rostest-test_launch_service_call.xml
[ERROR] [/service_adv] [1513807564.761673776]: client wants service /service_adv to have md5sum 6b193467bc2d41be616d6b0e985e2a63, but it has 671f8e4998eaec79f1c47e339dfd527b. Dropping connection.
[ERROR] [/service_adv] [1513807564.762505964]: client wants service /service_adv to have md5sum 6b193467bc2d41be616d6b0e985e2a63, but it has 671f8e4998eaec79f1c47e339dfd527b. Dropping connection.
[ERROR] [/service_adv] [1513807564.763258836]: client wants service /service_adv to have md5sum 6b193467bc2d41be616d6b0e985e2a63, but it has 671f8e4998eaec79f1c47e339dfd527b. Dropping connection.
[ERROR] [/service_adv] [1513807564.764014530]: client wants service /service_adv to have md5sum 6b193467bc2d41be616d6b0e985e2a63, but it has 671f8e4998eaec79f1c47e339dfd527b. Dropping connection.
[Testcase: testservice_call] ... ok

[ROSTEST]-----------------------------------------------------------------------

[test_roscpp.rosunit-service_call/callSrv][passed]
[test_roscpp.rosunit-service_call/callSrvUnicode][passed]
[test_roscpp.rosunit-service_call/callSrvMultipleTimes][passed]
[test_roscpp.rosunit-service_call/callSrvWithWrongType][passed]
[test_roscpp.rosunit-service_call/callSrvHandle][passed]
[test_roscpp.rosunit-service_call/callSrvPersistentHandle][passed]
[test_roscpp.rosunit-service_call/callSrvLongRunning][passed]
[test_roscpp.rosunit-service_call/callSrvLongRunningTimeout][passed]
[test_roscpp.rosunit-service_call/callSrvWhichUnadvertisesInCallback][passed]
[test_roscpp.rosunit-service_call/handleValid][passed]
[test_roscpp.rosunit-service_call/waitForServiceTimeout][passed]

SUMMARY
 * RESULT: SUCCESS
 * TESTS: 11
 * ERRORS: 0
 * FAILURES: 0

rostest log file is in /home/eperdomo/.ros/log/rostest-nb-eperdomo-19190.log
-- run_tests.py: verify result "/home/eperdomo/dev/ws/cpr_ws/build/test_roscpp/test_results/test_roscpp/rostest-test_launch_service_call.xml"
Built target run_tests_test_roscpp_rostest_test_launch_service_call.xml
```